src/pcre2.h.in

```diff
@@ -146,7 +146,7 @@ D   is inspected during pcre2_dfa_match() execution
 
 /* An additional compile options word is available in the compile context. */
 
-#define PCRE2_EXTRA_ALLOW_SURROGATE_ESCAPES  0x00000001u  /* C */
+#define PCRE2_EXTRA_ALLOW_SURROGATES         0x00000001u  /* C */
```

Comment:
It might be better to keep the old flag and add PCRE2_EXTRA_ALLOW_SURROGATES, which will imply it and extend it?

Reply:
Yes, I was already going to say that making the non-compatible change of removing an option is not something I'd want to do, especially as there is a way round it, as you suggest.

Reply:
... and I note that the GitHub tests have failed...

Reply:
correct, I posted this as a work in progress, to help gather feedback on the approach and caveats, and a roundabout way to ask Zoltan for help implementing the new behaviour (use a flag to redefine what is invalid utf), so that match_invalid_utf and utf behave similarly and resolve the ambiguity when matching \p{Cs} in the JIT, if it was sound and correct for the interpreter.

Will update the change and make the RFC more explicit.